wasmapi/go/helpers.go

```diff
@@ -49,6 +53,54 @@ func stringToBufPtr(s string) bufPtr {
 	return bufPtr(uint64(len(s))<<32 | uint64(uintptr(unsafePtr)))
 }
 
+// anytoBufPtr returns a bufPtr that encodes the pointer and length of the
+// input.
+// The input is first encoded to binary buffer, which is then used as the
+// returned bufPtr.
+// WARNING the binary encoding will only works on with fixed size data, *i.e.*
+// with int32 but not with int, as this data would be exchanged from 32 bits
+// WASM VM to host which can be 64 bits.
+func anytoBufPtr(a any) (bufPtr, error) {
+	buf := make([]byte, reflect.TypeOf(a).Size())
+	buffer := new(bytes.Buffer)
+
+	// TODO Use binary.Encode() once compiler no more yells about it:
+	// undefined: binary.Encode
+	err := binary.Write(buffer, binary.NativeEndian, a)
+	if err != nil {
+		return bufPtr(0), fmt.Errorf("converting %T to []byte: %w", a, err)
+	}
+
+	// TODO Remove this when using binary.Encode().
+	copy(buf, buffer.Bytes())
+
+	return bytesToBufPtr(buf), nil
+}
```

Comment:
> No, because the length of the buffer will only be the length of what is written while we want to have the length of the structure, even to have only final 0.

I see the binary package uses a [different way](https://cs.opensource.google/go/go/+/refs/tags/go1.23.2:src/encoding/binary/binary.go;l=680) than `reflect.TypeOf(a).Size()`, isn't it better to rely on the size returned by `binary.Write` than trying to calculate the size by ourselves?

Also, in what case do you think the written size could be smaller than the size of the structure?